File: StyleGuide-iOS/StyleGuide/Fonts/UIFont+Allinson.m
//
//  UIFont+Allinson.m
//  StyleGuide-iOS
//
//  Created by Chris Allinson on 2024-05-20.
//

#import "UIFont+Allinson.h"

@implementation UIFont (Allinson)

+ (UIFont *)Allinson_open_sans_small {
    UIFont *tempFont = [UIFont fontWithName:@"OpenSans" size:11.0];
    if (tempFont == NULL) {
        return [UIFont systemFontOfSize:11.0];
    } else {
        return tempFont;
    }
}

+ (UIFont *)Allinson_open_sans_medium {
    UIFont *tempFont = [UIFont fontWithName:@"OpenSans" size:13.0];
    if (tempFont == NULL) {
        return [UIFont systemFontOfSize:13.0];
    } else {
        return tempFont;
    }
}

+ (UIFont *)Allinson_open_sans_large {
    UIFont *tempFont = [UIFont fontWithName:@"OpenSans" size:15.0];
    if (tempFont == NULL) {
        return [UIFont systemFontOfSize:15.0];
    } else {
        return tempFont;
    }
}

+ (UIFont *)Allinson_open_sans_extralarge {
    UIFont *tempFont = [UIFont fontWithName:@"OpenSans" size:17.0];
    if (tempFont == NULL) {
        return [UIFont systemFontOfSize:17.0];
    } else {
        return tempFont;
    }
}


+ (UIFont *)Allinson_chalkduster_large {
    UIFont *tempFont = [UIFont fontWithName:@"Chalkduster" size:20.0];
    if (tempFont == NULL) {
        return [UIFont systemFontOfSize:20.0];
    } else {
        return tempFont;
    }
}

+ (UIFont *)Allinson_chalkduster_extralarge {
    UIFont *tempFont = [UIFont fontWithName:@"Chalkduster" size:40.0];
    if (tempFont == NULL) {
        return [UIFont systemFontOfSize:40.0];
    } else {
        return tempFont;
    }
}

@end

File: StyleGuide-iOS/StyleGuide/Fonts/UIFont+Allinson.swift
//
//  UIFont+Allinson.swift
//  Allinson iOS StyleGuide
//
//  Created by Chris Allinson on 2024-05-19.
//  Copyright © 2024 Chris Allinson. All rights reserved.
//

import UIKit

public extension UIFont {
    
    struct Allinson {
        
        // Allinson.ca fonts
        
        public static var open_sans_small: UIFont = UIFont(name: "OpenSans", size: 11.0) ?? UIFont.systemFont(ofSize: 11.0)
        public static var open_sans_medium: UIFont = UIFont(name: "OpenSans", size: 13.0) ?? UIFont.systemFont(ofSize: 13.0)
        public static var open_sans_large: UIFont = UIFont(name: "OpenSans", size: 15.0) ?? UIFont.systemFont(ofSize: 15.0)
        public static var open_sans_extralarge: UIFont = UIFont(name: "OpenSans", size: 17.0) ?? UIFont.systemFont(ofSize: 17.0)
        public static var chalkduster_large: UIFont = UIFont(name: "Chalkduster", size: 20.0) ?? UIFont.systemFont(ofSize: 20.0)
        public static var chalkduster_extralarge: UIFont = UIFont(name: "Chalkduster", size: 40.0) ?? UIFont.systemFont(ofSize: 40.0)
    }
}
